.test-infra/jenkins/CommonJobProperties.groovy

```diff
@@ -188,16 +189,9 @@ class CommonJobProperties {
 
     // Ensure that parallel workers don't exceed total available memory.
 
-    // TODO(BEAM-4230): OperatingSystemMXBeam incorrectly reports total memory; hard-code for now
-    // Jenkins machines are GCE n1-highmem-16, with 104 GB of memory
-    // def os = (com.sun.management.OperatingSystemMXBean)java.lang.management.ManagementFactory.getOperatingSystemMXBean()
-    // def totalMemoryMb = os.getTotalPhysicalMemorySize() / (1024*1024)
-    def totalMemoryMb = 104 * 1024
-    // Jenkins uses 2 executors to schedule concurrent jobs, so ensure that each executor uses only half the
-    // machine memory.
-    def totalExecutorMemoryMb = totalMemoryMb / 2
-    def perWorkerMemoryMb = totalExecutorMemoryMb / maxWorkers
-    context.switches("-Dorg.gradle.jvmargs=-Xmx${(int)perWorkerMemoryMb}m")
+    // For [BEAM-4847], hardcode Xms and Xmx to reasonable values (3g/8g).
```

Comment:
Won't we blow out the memory usage if we have two executors per machine.
24 workers * 8gbs > 104 gbs

Reply:
Yes, if the machine has 128GB then it won't be enough. I would propose to keep the JIRA open to also look for a non hardcoded solution perhaps?

Reply:
Only if all 24 workers are at max heap size? I'd expect a distribution among them where many are not max'd out.

Reply:
The JVM does not garbage collect eagerly and grows to the max heap size and then performs GC so its very likely that all workers will use max heap size if they run for any prolonged period of time.